### PR TITLE
Rename eon functions

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    - Rename `inShelleyBasedEraEonMaybe` to `forShelleyBasedEraInEonMaybe`
    - Rename `inShelleyBasedEraEon` to `forShelleyBasedEraInEon`
    - Rename `maybeEonInShelleyBasedEra` to `forShelleyBasedEraMaybeEon`
    - Rename `eonInShelleyBasedEra` to `inEonForShelleyBasedEra`
    - Rename `maybeEonInEra` to `forEraMaybeEon`
    - Rename `inEraEonMaybe` to `forEraInEonMaybe`
    - New `inEonForEraMaybe` function
    - New `inEonForShelleyBasedEraMaybe` function
# uncomment types applicable to the change:
  type:
  - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

The function names are composed on smaller positional components:

* `forEra` - corresponds to `CardanoEra era`
* `forShelleyBasedEra` - corresponds to `ShelleyBasedEra era`
* `inEon` - corresponds to `a` and `eon era -> a` if not coupled with `maybe` otherwise corresponds to just `eon era -> a`
* `maybe - corresponds to `Maybe a`
* `maybeEon` - corresponds to `Maybe (eon era)`

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]
I like the new naming better.

File: cardano-api/internal/Cardano/Api/Eon/ShelleyBasedEra.hs [hunk 1]
```diff
@@ -45,38 +46,46 @@ import qualified Data.Text as Text
...
 -- | Determine the value to use for a feature in a given 'ShelleyBasedEra'.
-eonInShelleyBasedEra :: ()
+inEonForShelleyBasedEra :: ()
   => Eon eon
   => a
   -> (eon era -> a)
   -> ShelleyBasedEra era
   -> a
-eonInShelleyBasedEra no yes =
+inEonForShelleyBasedEra no yes =
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/Eon/ShelleyBasedEra.hs:55]: why not `shelleyBasedEraInEon`?

File: cardano-api/internal/Cardano/Api/Eras/Core.hs [hunk 2]
```diff
@@ -125,19 +134,19 @@ forEraInEon :: ()
 forEraInEon era no yes =
   inEonForEra no yes era
 
-inEraEonMaybe :: ()
+forEraInEonMaybe :: ()
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/Eras/Core.hs:137]: :+1:

**Review by Jimbo4350 — APPROVED** [review]
:100: just one comment however

File: cardano-api/internal/Cardano/Api/Eon/ShelleyBasedEra.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-api/internal/Cardano/Api/Eon/ShelleyBasedEra.hs:55]: I'm using positional naming.

The `inEon` corresponds to this part of the type signature:

```
  => a
  -> (eon era -> a)
```

and the `ForShelleyBasedEra` corresponds to this part of the type signature:

```
  -> ShelleyBasedEra era
```

The words are in that order to match the order of the type signature to make it easy to remember.

**Pull request merged**